src/index.ts

```diff
@@ -1,69 +1,290 @@
 import {
   JupyterFrontEnd,
-  JupyterFrontEndPlugin
+  JupyterFrontEndPlugin,
+  ILayoutRestorer
 } from '@jupyterlab/application';
 import { deployAppIcon } from './icons';
-import { DocumentWidget } from '@jupyterlab/docregistry';
+import { WidgetTracker, IFrame, MainAreaWidget } from '@jupyterlab/apputils';
+import { Widget } from '@lumino/widgets';
+
+import { ISettingRegistry } from '@jupyterlab/settingregistry';
+
+const DEFAULT_BASE_URL = '/services/japps/create-app';
+
+const PLUGIN_ID = 'jupyterlab-jhub-apps:commands';
+
+/**
+ * Settings for the runtime configuration of the application.
+ * These are read from the settings registry and updated any
+ * time a user changes their settings.
+ */
+const runtimeSettings = {
```

Comment:
These are now global state. They are read from the settings registry at initialization. Then they are updated via `settings.changed.connect` whenever a user updates their settings.